07deBERTaClassifierForNewDataset.ipynb

```diff
@@ -0,0 +1,283 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Train our Baseline Model with New Dataset (DeBERTa)\n",
+    "\n",
+    "## Overview\n",
+    "\n",
+    "This notebook trains a classifier using the [DeBERTa small](https://huggingface.co/microsoft/deberta-v3-small/tree/main) model, applied to a new dataset generated by the Falcon 40B model from the ER5 tweet dataset. The data is formatted as a CSV file, from our methodology from the Bootcamp day 2 reference, 04ModelTrelevance.ipynb.\n",
+    "\n",
+    "## Key Steps and Objectives\n",
+    "\n",
+    "1. **Classifier Training**: We fine-tune the DeBERTa small model on the Falcon 40B dataset to adapt it to this specific classification task.\n",
+    "\n",
+    "2. **Results Visualization**: We analyze the model's performance through:\n",
+    "   - **Confusion Matrix**: Provides insights into accuracy and misclassification rates.\n",
+    "   - **ROC Curve**: Illustrates the model's performance across various thresholds.\n",
+    "\n",
+    "This notebook is essential for evaluating the DeBERTa-based classifier on the new dataset and offers comprehensive insights through detailed visualizations."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "plaintext"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "!jupyter kernelspec list"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "import transformers\n",
+    "import datasets\n",
+    "import functools\n",
+    "import os\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "import torch\n",
+    "import torch.nn.functional\n",
+    "import sklearn.metrics\n",
+    "import sklearn.model_selection\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "sys.path.append(\n",
+    "    \"/p/project/deepacf/maelstrom/haque1/AP2-Social-media-data-for-better-local-forecasts/bootcamp/AP2/scripts\"\n",
+    ")\n",
+    "import plotting"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "random_seed = 42\n",
+    "torch.manual_seed(random_seed)\n",
+    "np.random.seed(random_seed)\n",
+    "\n",
+    "# Load the dataset\n",
+    "folder_path = \"/p/project/deepacf/maelstrom/haque1/dataset/\"\n",
+    "file = \"tweets_2017_01_02_03_era5_normed_no_snow.csv\"\n",
+    "file_path = folder_path + file\n",
+    "ds_tweets = pd.read_csv(file_path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Again define labels\n",
+    "key_tp = \"content\"\n",
+    "ds_tweets[\"content\"] = ds_tweets[key_tp].astype(str)"
```

Comment:
```suggestion
    "key_input = \"content\"\n",
    "ds_tweets[\"content\"] = ds_tweets[key_input].astype(str)"
```
not talking about **T**otal **P**recipitation (=tp) here, variable names should reflect this